Source/bmalloc/libpas/src/libpas/pas_thread_local_cache_layout.c

```diff
@@ -71,6 +76,36 @@ pas_allocator_index pas_thread_local_cache_layout_add_node(pas_thread_local_cach
 
     result = pas_thread_local_cache_layout_next_allocator_index;
 
+    did_overflow = __builtin_add_overflow(
+        pas_thread_local_cache_layout_next_allocator_index,
+        pas_thread_local_cache_layout_node_num_allocator_indices(node),
+        &pas_thread_local_cache_layout_next_allocator_index);
+    PAS_ASSERT(!did_overflow);
+
+    begin_page = pas_round_up_to_power_of_2(pas_thread_local_cache_offset_of_allocator(result), page_size);
+    end_page = pas_round_up_to_power_of_2(pas_thread_local_cache_offset_of_allocator(pas_thread_local_cache_layout_next_allocator_index - 1), page_size);
+
+    /*
+     * Local allocator must not lie across multiple pages because pages can be decommited.
+     * Once we request decommit of pages, it depends on kernel's decision when these pages will be decommited.
+     * If local allocator lies across multiple pages, it is possible that some part of pages can be decommited,
+     * and we will get partially corrupt local allocator.
+     *
+     *                            decommited
+     *     [ page 1         ][  page 2         ]
+     *                [ local allocator ]
+     *                 ^
+     *                 Checking this bit says it is not decommited. But the latter part is already decommited.
+     */
+    if (begin_page != end_page) {
+        result = (begin_page - PAS_OFFSETOF(pas_thread_local_cache, local_allocators)) / sizeof(uint64_t);
```

Comment:
I hacked this, but will clean up later (this patch is draft).

Reply:
Also, I will check whether pas_scavenger can work well with these holes.

Reply:
Ugh, this requires some adjustment in designated heap indexes for x64 since x64 has 4KB page while ARM64 darwin has 16KB page (so, behavior is slightly different). I think for now, we should disable TLC decommit feature.